src/Controller/MessageController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\message_ui\Controller\MessageController.
+ */
+
+namespace Drupal\message_ui\Controller;
+
+use Drupal\Core\Url;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\message\Entity\Message;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\message\MessageTypeInterface;
+
+
+class MessageController extends ControllerBase implements ContainerInjectionInterface {
+
+  /**
+   * @var \Drupal\Core\Entity\EntityAccessControlHandlerInterface
+   */
+  private $accessHandler;
+
+  /**
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * Constructs a MessageUiController object.
+   */
+  public function __construct() {
+    $this->entityManager = \Drupal::entityManager();
+    $this->accessHandler = \Drupal::entityManager()->getAccessControlHandler('message');
+  }
+
+  /**
+   * Generates output of all message type entities with permission to create.
+   *
+   * @return array|\Symfony\Component\HttpFoundation\RedirectResponse
+   *   A render array for a list of the message types that can be added; however,
+   *   if there is only one node type defined for the site, the function
+   *   will return a RedirectResponse to the message.add_by_type page for that one message
+   *   type.
+   */
+  public function addPage() {
+    $build = [
+        '#theme' => 'message_add_list',
+    ];
+
+    $content = array();
+
+    // Only use node types the user has access to.
+    foreach ($this->entityManager()->getStorage('message_type')->loadMultiple() as $type) {
+      $access = $this->entityManager()->getAccessControlHandler('message')->createAccess($type->id(), NULL, [], TRUE);
+      if ($access->isAllowed()) {
+        $content[$type->id()] = $type;
+      }
+    }
+
+    // Bypass the admin/content/messages/create listing if only one content type is available.
+    if (count($content) == 1) {
+      $type = array_shift($content);
+      return $this->redirect('message_ui.add', array('message_type' => $type->id()));
+    }
+
+    $build['#content'] = $content;
+
+    if (!empty($build)) {
+      return $build;
```

Comment:
You can remove the `$build` array and do some thing like:
```php
return [
  ...
];
```

And not declare the variable.